lib/facter/dynflow.rb

```diff
@@ -0,0 +1,11 @@
+Facter.add(:foreman_dynflow) do
+  setcode do
+    directory = '/etc/foreman/dynflow'
+    extension = '.yml'
+    return unless File.exist?(directory)
+
+    Dir[File.join(directory, "*#{extension}")].map do |config|
+      File.basename(config, extension)
+    end
```

Comment:
```suggestion
    if File.exist?(directory)
      Dir[File.join(directory, "*#{extension}")].map do |config|
        File.basename(config, extension)
      end
    else
      {}
    end
```

To prevent annoying `Error when resolving custom fact: unexpected return` output in the case where it is being applied for the first time.

Reply:
Rather than `{}` I think you should return `nil`

Reply:
You're absolutely right. `nil` is what actually worked for me in my testing. I made a typo in the suggestion here.